cpp/oneapi/dal/algo/pca/backend/gpu/train_kernel_cov_impl_dpc.cpp

```diff
@@ -0,0 +1,730 @@
+            const std::int64_t local_size = item.get_local_range()[0];
+            const std::int64_t id = item.get_local_id()[0];
+            const std::int64_t group_id = item.get_group().get_id(0);
+
+            std::int64_t* lrc_ptr = lrc_buf.get_pointer().get();
+            Float* lsum_ptr = lsum_buf.get_pointer().get();
+            Float* lsum2cent_ptr = lsum2cent_buf.get_pointer().get();
+            Float* lmean_ptr = lmean_buf.get_pointer().get();
+
+            if (distr_mode) {
+                merge_blocks_kernel<Float, deffered_fin_true>(item,
+                                                              brc_ptr,
+                                                              bsum_ptr,
+                                                              bsum2cent_ptr,
+                                                              lrc_ptr,
+                                                              lsum_ptr,
+                                                              lmean_ptr,
+                                                              lsum2cent_ptr,
+                                                              rsum_ptr,
+                                                              rsum2cent_ptr,
+                                                              rmean_ptr,
+                                                              rvarc_ptr,
+                                                              id,
+                                                              group_id,
+                                                              local_size,
+                                                              block_count);
+            }
+            else {
+                merge_blocks_kernel<Float, deffered_fin_false>(item,
+                                                               brc_ptr,
+                                                               bsum_ptr,
+                                                               bsum2cent_ptr,
+                                                               lrc_ptr,
+                                                               lsum_ptr,
+                                                               lmean_ptr,
+                                                               lsum2cent_ptr,
+                                                               rsum_ptr,
+                                                               rsum2cent_ptr,
+                                                               rmean_ptr,
+                                                               rvarc_ptr,
+                                                               id,
+                                                               group_id,
+                                                               local_size,
+                                                               block_count);
+            }
+        });
```

Comment:
I strongly discourage you to use such constructions in kernels. 

It would be better to produce two variants of free function
`template <bool distributed, typename Float>
std::tuple<local_result<Float>, sycl::event> merge_blocks(...)`
and then use `if constexpr (distributed) {...` in kernel itself. 

In other case it can produce great bloat of binary code which is suboptimal for both compilers vectorization and relatively small instruction cache of each work group.